tests/e2e_tests/kgain_e2e.py

```diff
@@ -0,0 +1,125 @@
+    
+    ###### Setup necessary calibration files
+    # Create necessary calibration files
+    # we are going to make a new nonlinear calibration file using
+    # a combination of the II&T nonlinearity file and the mock headers from
+    # our unit test version of the NonLinearityCalibration
+    nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+    pri_hdr, ext_hdr = mocks.create_default_headers()
+    ext_hdr["DRPCTIME"] = time.Time.now().isot
+    ext_hdr['DRPVERSN'] =  corgidrp.__version__
+    mock_input_dataset = data.Dataset(mock_cal_filelist)
+    nonlinear_cal = data.NonLinearityCalibration(nonlin_dat,
+                                                 pri_hdr=pri_hdr,
+                                                 ext_hdr=ext_hdr,
+                                                 input_dataset=mock_input_dataset)
+    nonlinear_cal.save(filedir=kgain_outputdir, filename="mock_nonlinearcal.fits" )
+
+    # add calibration file to caldb
+    this_caldb = caldb.CalDB()
+    this_caldb.create_entry(nonlinear_cal)
+
+    ####### Run the walker on some test_data
+
+    for file in l1_data_filelist_same_exp:
+        image = data.Image(file)
+        # This should not be necessary anymore after the updates of the OBSTYPE keyword, up to now it is only "SCI"
+        if image.pri_hdr['OBSTYPE'] != 'MNFRAME':
+            image.pri_hdr['OBSTYPE'] = 'MNFRAME'
+            image.save(filename = file)
+        l1_data_filelist_range_exp.append(file)
+
+    walker.walk_corgidrp(l1_data_filelist_range_exp, "", kgain_outputdir, template="l1_to_kgain.json")
+    #kgain_files = glob.glob(os.path.join(kgain_outputdir, "CGI*.fits"))
+    #kgain_files.sort()
+    #print (kgain_files)
+    #dataset_kgain = data.Dataset(kgain_files)
+    kgain_file = os.path.join(kgain_outputdir, "CGI_EXCAM_L1_0000051731_kgain.fits")
+    kgain = data.KGain(kgain_file)
+    
+    # clean up by removing entry
+    this_caldb.remove_entry(nonlinear_cal)
+
+    #kgain = calibrate_kgain(dataset_kgain,
+    #                n_cal=3, n_mean=3, min_val=800, max_val=3000, binwidth=68,
+    #                make_plot=True,plot_outdir='figures', show_plot=True, 
+    #                logspace_start=-1, logspace_stop=4, logspace_num=200, 
+    #                verbose=True, detector_regions=None)
+    
+    ##### Check against TVAC kgain, readnoise
+    new_kgain = kgain.value
+    new_readnoise = kgain.ext_hdr["RN"]
+    print("determined kgain:", new_kgain)
+    print("determined read noise", new_readnoise)    
+    
+    diff_kgain = new_kgain - tvac_kgain
+    diff_readnoise = new_readnoise - tvac_readnoise
+    print ("difference to TVAC kgain:", diff_kgain)
+    print ("difference to TVAC read noise:", diff_readnoise)
+    print ("error of kgain:", kgain.error)
+    print ("error of  readnoise:", kgain.ext_hdr["RN_ERR"])
+
+    assert np.abs(diff_kgain) < 1
```

Comment:
are these differences small enough to be tolerated? We should try running on the full dataset and circle back to this.

Reply:
We could use the determined errors to define a tolerance.

Reply:
Understood, so then just a very small number